### ISSUE TITLE
Celery deactivated instead of error code 1

### ISSUE THREAD

**Issue by yvovandoorn** [pretalx/pretalx#1419]
### Problem and impact

Recently a Redis upgrade forced the node Celery was connected to from `master` to `replica`. This resulted in the following non-recoverable error:

```
Jan 19 09:17:21 ip-10-0-0-27 celery[488]: [2023-01-19 09:17:21,504: CRITICAL/MainProcess] Unrecoverable error: ResponseError('UNBLOCKED force unblock from blocking operation, instance state changed (master -> replica?)')
Jan 19 09:17:21 ip-10-0-0-27 celery[488]: Traceback (most recent call last):
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/celery/worker/worker.py", line 203, in start
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     self.blueprint.start(self)
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/celery/bootsteps.py", line 116, in start
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     step.start(parent)
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/celery/bootsteps.py", line 365, in start
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     return self.obj.start()
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/celery/worker/consumer/consumer.py", line 332, in start
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     blueprint.start(self)
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/celery/bootsteps.py", line 116, in start
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     step.start(parent)
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/celery/worker/consumer/consumer.py", line 628, in start
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     c.loop(*c.loop_args())
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/celery/worker/loops.py", line 97, in asynloop
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     next(loop)
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/kombu/asynchronous/hub.py", line 362, in create_loop
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     cb(*cbargs)
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/kombu/transport/redis.py", line 1326, in on_readable
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     self.cycle.on_readable(fileno)
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/kombu/transport/redis.py", line 562, in on_readable
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     chan.handlers[type]()
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/kombu/transport/redis.py", line 955, in _brpop_read
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     dest__item = self.client.parse_response(self.client.connection,
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/redis/client.py", line 1084, in parse_response
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     response = connection.read_response()
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:   File "/app/pretalx/.local/lib/python3.10/site-packages/redis/connection.py", line 765, in read_response
Jan 19 09:17:21 ip-10-0-0-27 celery[488]:     raise response
Jan 19 09:17:21 ip-10-0-0-27 celery[488]: redis.exceptions.ResponseError: UNBLOCKED force unblock from blocking operation, instance state changed (master -> replica?)
Jan 19 09:17:23 ip-10-0-0-27 systemd[1]: pretalx-worker.service: Deactivated successfully.
Jan 19 09:17:23 ip-10-0-0-27 systemd[1]: pretalx-worker.service: Consumed 10min 32.471s CPU time.
```

Specifically `Jan 19 09:17:23 ip-10-0-0-27 systemd[1]: pretalx-worker.service: Deactivated successfully.` shows that the process didn't exit with `error code 1`. 

### Expected behaviour

I would expect the process to exit with a FAILURE, thus allowing `systemd` to do its work if the systemd unit file for pretalx has a restart upon failure option. Hypothetically the following should've happened:

```
Jan 19 09:17:21 ip-10-0-0-27 celery[488]: [2023-01-19 09:17:21,504: CRITICAL/MainProcess] Unrecoverable error: ResponseError('UNBLOCKED force unblock from blocking operation, instance state changed (master -> replica?)')
Jan 19 09:17:21 ip-10-0-0-27 systemd[1]: pretalx-worker.service: Main process exited, code=exited, status=1/FAILURE
Jan 19 09:17:21 ip-10-0-0-27 systemd[1]: pretalx-worker.service: Failed with result 'exit-code'.
Jan 19 09:17:21 ip-10-0-0-27 systemd[1]: pretalx-worker.service: Consumed 10min 32.471s CPU time.
Jan 19 09:17:21 ip-10-0-0-27 systemd[1]: pretalx-worker.service: Scheduled restart job, restart counter is at 1.
Jan 19 09:17:21 ip-10-0-0-27 systemd[1]: Stopped pretalx background worker.
Jan 19 09:17:21 ip-10-0-0-27 systemd[1]: pretalx-worker.service: Consumed 10min 32.471s CPU time.
Jan 19 09:17:21 ip-10-0-0-27 systemd[1]: Started pretalx background worker.
```

### Steps to reproduce

1. Force a failure within Redis to go to `replica`
2. Watch pretalx worker 'gracefully' deactivate itself
3. No more emails sent

### Version

Current main

**Comment by rixx** [comment]
Is this something to be fixed in pretalx (and if so: how)? Your issue description reads as if this is something that is either a Celery bug (possible, though I'd guess not terribly likely), a conscious Celery design choice, or something to be fixed with configuration / CLI args.

Tentatively closing as out-of-scope, but will happily reopen if it turns out that I'm wrong!

**Comment by yvovandoorn** [comment]
@rixx so I did a bit more research on this. It does appear that Celery itself has changed behavior. In 4.x (which my `pretix` stack is still running) it was seen as a failure but in 5.x (which `pretalx` is running) it appears to disable. I suspect it has something to do with the introduction of 5.0's connection retry logic, which is unconfigured when launching Celery. 

Either way, with https://github.com/pretix/pretix/pull/2983, its possible that Pretix will behave in the future until https://github.com/celery/celery/issues/7946 makes it into a future (5.3?) version of Celery. 

I'm running both `pretix` and `pretalx` with the same Celery backend and just found it odd that with my `pretix` setup, the daemon `pretix-worker` saw it as a failure (and thus crashed, and recovered a second later by hosts' systemd configuration. Whereas the `pretalx-worker` process deactivated itself. Now the `pretix` setup is running an older version of celery -- `4.4.7` whereas `pretalx` is using `5.2.7`. 

So you could be right that this out side of the scope for
